source/tutorials/connect/compression.txt

```diff
@@ -0,0 +1,136 @@
+.. _javars-compression:
+
+===========
+Compression
+===========
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+The {+driver-short+} supports compression of messages to and from MongoDB
+servers. The driver implements the three algorithms that are supported
+by MongoDB servers:
+
+- `Snappy <https://google.github.io/snappy/>`__: Snappy compression can
+  be used when connecting to MongoDB servers running version 3.4 and later.
+- `Zlib <https://zlib.net/>`__: Zlib compression can be used when
+  connecting to MongoDB servers running version 3.6 and later.
+- `Zstandard <https://github.com/facebook/zstd/>`__: Zstandard
+  compression can be used when connecting to MongoDB servers running
+  version 4.2 and later.
+
+The driver will negotiate which, if any, compression algorithm is used
+based on capabilities advertised by the server in the ``hello`` command
+response.
+
+Specify Compression By Using ConnectionString
+---------------------------------------------
+
+Include the following import statements:
+
+.. code-block:: java
+
+   import com.mongodb.ConnectionString;
+   import com.mongodb.reactivestreams.client.MongoClients;
+   import com.mongodb.reactivestreams.client.MongoClient;
+
+To specify compression within a ``ConnectionString``, specify
+compressors as part of the connection string.
+
+The following code specifies the Snappy compression algorithm:
+
+.. code-block:: java
+
+   ConnectionString connectionString = new ConnectionString("mongodb://localhost/?compressors=snappy");
```

Comment:
Suggestion:
The code block formatting only shows up to "localhost/?" on the first line of each of these. Would it be possible to format on a new line so it is not necessary to scroll horizontally to see the difference?